contracts/KinFoundationVestingTrustee.sol

```diff
@@ -0,0 +1,230 @@
+    address public kinFoundation;
+
+    // Funds already transferred to Kin Foundation.
+    uint256 public transferred = 0;
+
+    // Vesting start time.
+    uint256 public startTime;
+
+    // Total vesting length.
+    uint256 public TOTAL_VESTING_TIME = 60 years;
+
+    // Annual grants for Kin Foundation. Each member index represents the amount of KIN granted in that year e.g.
+    // member #0 is the funds vested for the first year, and member #2 is for the third year.
+    //
+    // For more information see the Kin whitepaper, chapter 6.
+    uint256[60] public ANNUAL_GRANTS = [
+        1.2e30,
+        9.6e29,
+        7.68e29,
+        6.144e29,
+        4.9152e29,
+        3.93216e29,
+        3.145728e29,
+        2.5165824e29,
+        2.01326592e29,
+        1.610612736e29,
+        1.2884901888e29,
+        1.03079215104e29,
+        8.24633720832e28,
+        6.597069766656e28,
+        5.2776558133248e28,
+        4.22212465065984e28,
+        3.377699720527872e28,
+        2.7021597764222976e28,
+        2.16172782113783808e28,
+        1.729382256910270464e28,
+        1.3835058055282163712e28,
+        1.10680464442257309696e28,
+        8.85443715538058477568e27,
+        7.083549724304467820544e27,
+        5.6668397794435742564352e27,
+        4.53347182355485940514816e27,
+        3.626777458843887524118528e27,
+        2.9014219670751100192948224e27,
+        2.32113757366008801543585792e27,
+        1.856910058928070412348686336e27,
+        1.485528047142456329878949068e27,
+        1.188422437713965063903159255e27,
+        9.50737950171172051122527404e26,
+        7.60590360136937640898021923e26,
+        6.08472288109550112718417538e26,
+        4.86777830487640090174734031e26,
+        3.89422264390112072139787225e26,
+        3.1153781151208965771182978e26,
+        2.49230249209671726169463824e26,
+        1.99384199367737380935571059e26,
+        1.59507359494189904748456847e26,
+        1.27605887595351923798765478e26,
+        1.02084710076281539039012382e26,
+        8.1667768061025231231209906e25,
+        6.5334214448820184984967924e25,
+        5.226737155905614798797434e25,
+        4.1813897247244918390379472e25,
+        3.3451117797795934712303577e25,
+        2.6760894238236747769842862e25,
+        2.1408715390589398215874289e25,
+        1.7126972312471518572699432e25,
+        1.3701577849977214858159545e25,
+        1.0961262279981771886527636e25,
+        8.769009823985417509222109e24,
+        7.015207859188334007377687e24,
+        5.61216628735066720590215e24,
+        4.48973302988053376472172e24,
+        3.591786423904427011777376e24,
+        2.873429139123541609421901e24,
+        1.1493716556494166437687604e25
+    ];
+
+    // The frequency by which the grant will vest.
+    uint256 public constant VESTING_FREQUENCY = 1 days;
+
+    // Used for calculating daily vested funds in calculateVestedTokens().
+    uint256 private constant DAYS_IN_YEAR = 365;
```

Comment:
Do you need to take leap years into account? Or will the noise be negligible?

Reply:
I doubt people will mind that the grant is unlocked one day earlier every 1460 days :). Alternatively, reduce the annual grants in the leap year indices by a factor of 1/366 (0.273%)

Reply:
So after multiplying every leap year by (365/366) and putting the excess in the final index, I get the schedule below. Notice how the final grant is two orders of magnitude larger than in the original schedule. This is because of the relatively larger error in year 3 (the first leap year, 2020).


1.200000000000000000E+30
--
9.600000000000000000E+29
7.659016393442620000E+29
6.144000000000000000E+29
4.915200000000000000E+29
3.932160000000000000E+29
3.137133114754100000E+29
2.516582400000000000E+29
2.013265920000000000E+29
1.610612736000000000E+29
1.284969723803280000E+29
1.030792151040000000E+29
8.246337208320000000E+28
6.597069766656000000E+28
5.263235988698230000E+28
4.222124650659840000E+28
3.377699720527870000E+28
2.702159776422290000E+28
2.155821460970790000E+28
1.729382256910270000E+28
1.383505805528210000E+28
1.106804644422570000E+28
8.830244704136370000E+27
7.083549724304460000E+27
5.666839779443570000E+27
4.533471823554850000E+27
3.616868230814250000E+27
2.901421967075110000E+27
2.321137573660080000E+27
1.856910058928070000E+27
1.481469227341510000E+27
1.188422437713960000E+27
9.507379501711720000E+26
7.605903601369370000E+26
6.068097955190870000E+26
4.867778304876400000E+26
3.894222643901120000E+26
3.115378115120890000E+26
2.485492922446170000E+26
1.993841993677370000E+26
1.595073594941890000E+26
1.276058875953510000E+26
1.018057901033950000E+26
8.166776806102520000E+25
6.533421444882010000E+25
5.226737155905610000E+25
4.169965162635080000E+25
3.345111779779590000E+25
2.676089423823670000E+25
2.140871539058930000E+25
1.708017730615330000E+25
1.370157784997720000E+25
1.096126227998170000E+25
8.769009823985410000E+24
6.996040624600380000E+24
5.612166287350660000E+24
4.489733029880530000E+24
3.591786423904420000E+24
2.865578239836320000E+24
3.565622171859610000E+27

Reply:
Nice finding @ryana , good job! :+1: 

Also @v1thesource thanks for the simulation. I agree with you - we'll stick with the original.